tests/Unit/NumericalAlgorithms/Spectral/Test_SwshDerivatives.cpp

```diff
@@ -300,6 +300,49 @@ void test_compute_angular_derivatives() noexcept {
   }
 }
 
+template <typename InverseDerivativeKind, typename DerivativeKind,
+          ComplexRepresentation Representation, int Spin>
+void test_inverse_derivative() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> size_distribution{2, 7};
+  const size_t l_max = size_distribution(gen);
+  const size_t number_of_radial_points = 2;
+  UniformCustomDistribution<double> coefficient_distribution{0.1, 1.0};
+
+  // fill the expected collocation point data by evaluating the analytic
+  // functions. This is very slow and imprecise (due to factorial division), but
+  // comparatively simple to formulate.
+  SpinWeighted<ComplexModalVector, Spin> generated_modes{
+      size_of_libsharp_coefficient_vector(l_max) * number_of_radial_points};
+  TestHelpers::generate_swsh_modes<Spin>(
+      make_not_null(&generated_modes.data()), make_not_null(&gen),
+      make_not_null(&coefficient_distribution), number_of_radial_points, l_max);
+
+  const auto computed_collocation =
+      inverse_swsh_transform(l_max, number_of_radial_points, generated_modes);
+  const auto expected_derivative_collocation =
+      inverse_swsh_transform(l_max, number_of_radial_points, generated_modes);
+
+  // perform inverse derivative operator
+  const auto inverse_derivative_values =
+      angular_derivative<InverseDerivativeKind, Representation>(
+          l_max, number_of_radial_points, computed_collocation);
+
+  // perform forward derivative operator
```

Comment:
If I understand correctly, this test checks the derivative and the inverse cancel, but doesn't check the derivative itself. Is there no way to test that, perhaps by taking a derivative of a function you know analytically?